subprojects/docs/src/docs/userguide/what_is_gradle.adoc

```diff
@@ -12,133 +12,114 @@
-
- * Custom task actions.
-+
-You can attach custom build logic that executes before or after a task via the link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doFirst(org.gradle.api.Action)[Task.doFirst()] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doLast(org.gradle.api.Action)[Task.doLast()] methods.
+=== Builds
 
- * <<writing_build_scripts#sec:extra_properties,Extra properties>> on projects and tasks.
-+
-These allows you to add your own properties to a project or task that you can then use from your own custom actions or any other build logic. Extra properties can even be applied to tasks that aren't explicitly created by you, such as those created by Gradle's core plugins.
+A build is an execution of a collection of tasks in a Gradle project. You instantiate a build with a Gradle command
+via the CLI or an IDE. In the command, you specify a task or tasks. Gradle determines the complete set of tasks to run based
+on inter-task dependencies, and runs the smallest set of tasks required to complete the specified task and its dependencies.
 
- * Custom conventions.
-+
-Conventions are a powerful way to simplify builds so that users can understand and use them more easily. This can be seen with builds that use standard project structures and naming conventions, such as <<building_java_projects#building_java_projects,Java builds>>. You can write your own plugins that provide conventions — they just need to configure default values for the relevant aspects of a build.
+== Optimizations
 
- * <<implementing_gradle_plugins.adoc#modeling_dsl_like_apis,A custom model>>.
-+
-Gradle allows you to introduce new concepts into a build beyond tasks, files and dependency configurations. You can see this with most language plugins, which add the concept of <<building_java_projects#sec:java_source_sets,_source sets_>> to a build. Appropriate modeling of a build process can greatly improve a build's ease of use and its efficiency.
+Over the years, Gradle has engineered a number of ways to make your builds faster.
+Gradle enables some of these, like the Daemon, by default. You must explicitly enable some other optimizations
+through configuration or build logic. This section highlights some of the most important
+optimizations to know about when you start using Gradle.
 
+=== Configuration Cache
 
-=== 5. Build scripts operate against an API
+The configuration phase is a large up-front cost for all Gradle builds. Every time you build your project, Gradle
+configures the project based on available inputs like build scripts, settings, and system properties. The
+configuration cache allows Gradle to cache the result of this configuration. This can save significant time
+in every build. However, the configuration cache is only compatible with certain syntax.
```

Comment:
Since this isn't production ready yet, we may not want to mention this yet or at least not first.

Reply:
Considering the configuration cache is already a concern frequently raised in plugin design, blog posts, and elsewhere in the docs, I figured it was a good idea to raise it here. Open to removing it if you don't agree, but I think the cat is out of the bag.